src/components/playbackSettings/playbackSettings.js

```diff
@@ -247,6 +257,33 @@ import template from './playbackSettings.template.html';
         });
     }
 
+    function onMaxVideoWidthChange(e) {
+        const context = this.options.element;
+
+        const selectVideoInNetworkQuality = context.querySelector('.selectVideoInNetworkQuality');
+        const selectVideoInternetQuality = context.querySelector('.selectVideoInternetQuality');
+        const selectChromecastVideoQuality = context.querySelector('.selectChromecastVideoQuality');
+
+        const selectVideoInNetworkQualityValue = selectVideoInNetworkQuality.value;
+        const selectVideoInternetQualityValue = selectVideoInternetQuality.value;
+        const selectChromecastVideoQualityValue = selectChromecastVideoQuality.value;
+
+        const maxVideoWidth = parseInt(e.target.value || '0', 10) || 0;
+
+        fillQuality(selectVideoInNetworkQuality, true, 'Video', maxVideoWidth);
+        fillQuality(selectVideoInternetQuality, false, 'Video', maxVideoWidth);
+        fillChromecastQuality(selectChromecastVideoQuality, maxVideoWidth);
+
+        selectVideoInNetworkQuality.value = selectVideoInNetworkQualityValue;
+        selectVideoInternetQuality.value = selectVideoInternetQualityValue;
+        selectChromecastVideoQuality.value = selectChromecastVideoQualityValue;
+
+        // HACK: Fix wrong value if it out of list
+        selectVideoInNetworkQuality.value = selectVideoInNetworkQuality.value;
+        selectVideoInternetQuality.value = selectVideoInternetQuality.value;
+        selectChromecastVideoQuality.value = selectChromecastVideoQuality.value;
```

Comment:
SonarCloud doesn't like this. Any ideas?

Reply:
I'm not sure I understand what this is supposed to be doing :sweat_smile:

Reply:
If the old value is not found in the new list, select becomes empty (its value is undefined). This block sets the existing or undefined value back, 
to fill select.

Reply:
Just so I am clear, setting the value the first time does not accomplish this then?

Reply:
Yes, at least not the way I want. If I set a value that is not in the list, select does not choose default (with undefined `value` attribute - `Auto`), it just becomes empty.

Reply:
Now also Lint is not satisfied :man_facepalming:

Reply:
What do you think about doing something like this to avoid the hack?

```js
document.querySelector('.selectLabelMaxVideoWidth option[value="0"]')?.selected = true; 
```

Reply:
I only need to do this if the old value is not in the list. I found solution.